### PR TITLE
[LLM] - [] - Upgrade legacy components to modern components

### PR DESCRIPTION
**Pull request by sshmaxime**
### 📝 Description

This PR subscribe itself in the long term improvement of this repository. It doesn't change anything on the user side but rather improve the developer experience and the codebase step by step.

**Files with high risk changes:**

[SetEnvsFromSettings.ts](https://github.com/LedgerHQ/ledger-live/pull/5224/files#diff-f7d8c87e1340cbb323a14e894efff2cbdff7dc5304971874e1fcd43875d09e55)

### ❓ Context

- **Impacted projects**: `LLM` 
- **Linked resource(s)**: [LIVE-10018] 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-10018]: https://ledgerhq.atlassian.net/browse/LIVE-10018?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/ReceiveFunds/03a-ConnectDevice.tsx [hunk 1]
```diff
@@ -144,7 +144,7 @@ export default function ConnectDevice({
   }
 
   if (!mainAccount.freshAddress) {
-    return <NotSyncedWarning continue={onSkipDevice} accountId={mainAccount.id} />;
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/screens/ReceiveFunds/03a-ConnectDevice.tsx:147]: Never used on the other component.

File: apps/ledger-live-mobile/src/screens/SendFunds/04-Summary.tsx [hunk 2]
```diff
@@ -371,14 +371,4 @@ const styles = StyleSheet.create({
   },
 });
 
-// FIXME: PROBABLY SOME TYPE OF StyleProp<ViewStyle>
-class VerticalConnector extends Component<{
```
Comment (sshmaxime) [inline apps/ledger-live-mobile/src/screens/SendFunds/04-Summary.tsx:375]: Idk why that was created in the first place.

File: apps/ledger-live-mobile/src/screens/ImportAccounts/DisplayResultItem.tsx [hunk 3]
```diff
@@ -56,3 +19,38 @@ const styles = StyleSheet.create({
...
+type Props = {
+  account: Account;
+  mode: ImportItem["mode"];
+  checked: boolean;
+  importing: boolean;
+  onSwitch: (checked: boolean, account: Account) => void;
+};
+
+const DisplayResultItem = ({ account, mode, checked, importing, onSwitch }: Props) => {
+  const _onSwitch = () => {
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/screens/ImportAccounts/DisplayResultItem.tsx:32]: Can you rename it without _, please? As _ prefix is a convention for private functions/ attributes.
You can rename it as `onSwitchAccount` or `handleSwitch` etc...

File: apps/ledger-live-mobile/src/components/SetEnvsFromSettings.ts [hunk 4]
```diff
@@ -1,45 +1,25 @@
...
-
-  componentDidMount() {
-    this.apply();
-  }
+  };
 
-  componentDidUpdate() {
-    this.apply();
-  }
+  useEffect(() => apply());
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/components/SetEnvsFromSettings.ts:20]: Don't we want to setEnvUnsafe when `hideEmptyTokenAccountsEnabled` or `filterTokenOperationsZeroAmountEnabled` changed ?
I think that we could have one useEffect for both with each value in the dependency array, wdyt?

**Review by KVNLS — CHANGES_REQUESTED** [review]

File: apps/ledger-live-mobile/src/components/SetEnvsFromSettings.ts (hunk 4 above)
Comment (sshmaxime) [reply apps/ledger-live-mobile/src/components/SetEnvsFromSettings.ts:20]: Yeah that'd make more sense !

**Review by KVNLS — APPROVED** [review]
LGTM!

**Pull request merged**